apps/src/StudioApp.js

```diff
@@ -1813,6 +1816,23 @@ function runButtonClickWrapper(callback) {
     $(window).trigger('run_button_pressed');
     $(window).trigger('appModeChanged');
   }
+
+  // inform Blockly that the run button has been pressed
+  if (window.Blockly && Blockly.mainBlockSpace) {
+    var customEvent;
+    // createEvent functionality is officially deprecated in favor of
+    // the event constructor, but some older browsers do not yet support
+    // event constructors. Attempt to use the new functionality, fall
+    // back to the old if it fails.
+    try {
+      customEvent = new Event(Blockly.BlockSpace.EVENTS.RUN_BUTTON_CLICKED);
+    } catch (e) {
+      customEvent = document.createEvent('Event');
+      customEvent.initEvent(Blockly.BlockSpace.EVENTS.RUN_BUTTON_CLICKED, true, false);
+    }
+    Blockly.mainBlockSpace.getCanvas().dispatchEvent(customEvent);
```

Comment:
Fix LGTM!

One nit, can take or leave, feels like maybe a tiny bit too much logic living in this portion of the code. Couple of ideas:

* add as `./utils.js` method, so `utils.createEvent(Blockly.BlockSpace.EVENTS.RUN_BUTTON_CLICKED)`
* if something in Google Closure tools / blockly utils supports this sort of dispatching and deals with the x-browser issue under the hood (do other events get triggered similarly?), another option could be to move this portion of the logic down in to blockly-core and use a closure tools helper. So we then might call e.g., `Blockly.mainBlockSpace.dispatchRunButtonClicked();`